DDF-1393 Add anonymous user (base security) configuration to installer

clockard opened the pull request:
- Added installer page after general configurations for anon claims handler
 - Added extension to ConfigurationAdmin to support new functionality
 - Removed 'uid' entry

@stustison @andrewkfiedler @djblue

<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/codice/ddf/124)
<!-- Reviewable:end -->

stustison commented on platform/admin/ui/src/main/webapp/js/views/installer/AnonClaims.view.js:
```diff
@@ -0,0 +1,414 @@
+        itemView: AnonClaimsMultiValuedEntry,
+        tagName: 'div',
+        modelEvents: {
+            "change": "render"
```
You shouldn't need the model event here. A collection view will automatically add and remove items from its collection.

clockard replied:
Removed the modelEvent. Waiting for everyone to finish reviewing before I push my changes up.

stustison commented on platform/admin/ui/src/main/webapp/templates/installer/anonClaims.handlebars:
```diff
@@ -0,0 +1,33 @@
+            <div id="claims-profiles"></div>
+
+            <!--</form>-->
+            <form class="form-horizontal add-federated-source">
```
Make sure you remove leftover styles when you're copying and pasting ;)

clockard replied:
Removed the unused 'add-federated-source' class. Waiting for everyone to finish reviewing before I push my changes up.

clockard commented:
@djblue has volunteered to hero this pr.

clockard commented:
To review the screenshots go to https://invis.io/RS3XUNCUV. If you leave comments there please make a note in this PR.

andrew-fiedler commented:
Review status: 0 of 21 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

<sup>**[platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/AnonClaimsHandlerExt.java, line 64 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/124#-JxG1uXwMI03IM0AEwvD)** ([raw file](https://github.com/codice/ddf/blob/4b84389d90e71ebe9b5f0aef0a7a7dfa53abf9da/platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/AnonClaimsHandlerExt.java#L64)):</sup>
You can return the evaluation of ```name.endsWith(".properties")``` directly.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/124)
<!-- Sent from Reviewable.io -->

clockard commented:
Review status: 0 of 21 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

<sup>**[platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/AnonClaimsHandlerExt.java, line 64 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/124#-JxG1uXwMI03IM0AEwvD)** ([raw file](https://github.com/codice/ddf/blob/4b84389d90e71ebe9b5f0aef0a7a7dfa53abf9da/platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/AnonClaimsHandlerExt.java#L64)):</sup>
Good call. I will fix it.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/124)
<!-- Sent from Reviewable.io -->

stustison commented:
Review status: 0 of 21 files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

<sup>**[platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/AnonClaimsHandlerExt.java, line 35 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/124#-JxVAwLZZXcjXTREool0)** ([raw file](https://github.com/codice/ddf/blob/4b84389d90e71ebe9b5f0aef0a7a7dfa53abf9da/platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/AnonClaimsHandlerExt.java#L35)):</sup>
Don't use XLogger, we use slf4j.

---

<sup>**[platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/AnonClaimsHandlerExt.java, line 85 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/124#-JxVALqwevsyPvZK7yU0)** ([raw file](https://github.com/codice/ddf/blob/4b84389d90e71ebe9b5f0aef0a7a7dfa53abf9da/platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/AnonClaimsHandlerExt.java#L85)):</sup>
This is not the correct way to iterate over a map. Use map.entrySet() and get the key and value from the entry.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/124)
<!-- Sent from Reviewable.io -->

clockard commented:
Review status: 0 of 21 files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

<sup>**[platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/AnonClaimsHandlerExt.java, line 35 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/124#-JxVAwLZZXcjXTREool0)** ([raw file](https://github.com/codice/ddf/blob/4b84389d90e71ebe9b5f0aef0a7a7dfa53abf9da/platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/AnonClaimsHandlerExt.java#L35)):</sup>
Fixed

---

<sup>**[platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/AnonClaimsHandlerExt.java, line 85 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/124#-JxVALqwevsyPvZK7yU0)** ([raw file](https://github.com/codice/ddf/blob/4b84389d90e71ebe9b5f0aef0a7a7dfa53abf9da/platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/AnonClaimsHandlerExt.java#L85)):</sup>
Fixed

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/124)
<!-- Sent from Reviewable.io -->

brendan-hofmann commented on platform/admin/ui/src/main/webapp/templates/installer/anonClaims.handlebars:
```diff
@@ -0,0 +1,33 @@
+            <!--</form>-->
+            <form class="form-horizontal">
+                <fieldset>
+                    <span>These are the security claims attributes that will be given to anonymous users. Additionally all other users will also be given these attributes in addition to there own.</span>
```
Typo, 'there own' should be 'their own'.

clockard replied:
Fixing

brendan-hofmann commented on platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/AnonClaimsHandlerExt.java:
```diff
@@ -0,0 +1,146 @@
+
+    private List<String> availableClaims;
+
+    //holds it name of claims that if modified would cause anonymous login failures
```
"holds it name"? If 'it' is an acronym, maybe that should be spelled out, otherwise this is probably a typo?

clockard replied:
Should say 'holds names' will fix

lessarderic commented:
Review status: 0 of 24 files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

<sup>**[platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/AnonClaimsHandlerExt.java, line 32 \[r5\]](https://reviewable.io:443/reviews/codice/ddf/124#-Jxf6Y-kS2E2hFoY1nar)** ([raw file](https://github.com/codice/ddf/blob/13e14172c9435a46dd008c551ccef68fd11764ca/platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/AnonClaimsHandlerExt.java#L32)):</sup>
Class comments would be nice. And what does Ext mean?

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/124)
<!-- Sent from Reviewable.io -->

clockard commented:
Review status: 0 of 24 files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

<sup>**[platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/AnonClaimsHandlerExt.java, line 32 \[r5\]](https://reviewable.io:443/reviews/codice/ddf/124#-Jxf6Y-kS2E2hFoY1nar)** ([raw file](https://github.com/codice/ddf/blob/13e14172c9435a46dd008c551ccef68fd11764ca/platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/AnonClaimsHandlerExt.java#L32)):</sup>
Adding the following class comment
/**
 * AnonClaimsHandlerExt is an extension for the ConfigurationAdminMBean
 * that adds capabilities to read/handle claim and profiles for the
 * anonymous claims handler.
*/

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/124)
<!-- Sent from Reviewable.io -->

djblue commented on platform/admin/ui/src/main/webapp/less/installer/style.less:
```diff
@@ -135,4 +142,108 @@ center {
+    .claim-button;
+}
+
+.row-container-div{
```
I don't think we need this class anymore.

clockard replied:
Removed unused class

djblue commented:
:+1:
